### PR TITLE
chore: update contributor guidelines

### PR DESCRIPTION
**Pull request by edalex-yinzi**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [x] documentation is changed or added

##### Description of change

Add more information to `setting up development environment ` section in contributor guidelines.
Change the order of `IDE` section.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -225,7 +288,101 @@ have changes automatically bundled up and reloaded in the browser:
...
+It can be done with the Java Control Panel or directly adding the domain in a new line in this file:
+\${user.home}/.java/deployment/security/exception.sites
+
+## Building the installer
+
+This guide runs sbt in non-interactive mode. You can run in interactive mode to save rebuild time by first running 'sbt', and the another command such as 'compile'.
+
+```bash
+cd to the {Equella repo} directory
+~$ sbt installerZip
```
Comment (edalex-ian) [inline CONTRIBUTING.md:382]: Two things:

- good to stay consistent with paths; and
- all the uses of SBT should be updated in this file to use the wrapper via `./sbt` (there's several above) and we can drop the shell prompt for consistency me thinks.

So as well as this below suggestion, could you please go and update the above uses of sbt too?

```suggestion
cd /path/to/openEquella
./sbt installerZip
```

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Nice work @edalex-yinzi . Definitely getting there. :wink: 

Please make the suggested changes, and then grab me tomorrow to discuss how to merge the JS/New UI sections.

File: CONTRIBUTING.md (hunk 1 above)
Comment (edalex-yinzi) [reply CONTRIBUTING.md:382]: Sure.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -115,8 +116,9 @@ Please follow the installation instructions provided by the tool.
 
 ### Install Java
 At the time of writing the openEQUELLA is built with Java 8. For this, we prefer to use
-openJDK and that latest version of that is 8.0.302. To install that with SDKMAN:
-```shell
+openJDK which is current at version 8.0.302. To install that with SDKMAN:
```
Comment (edalex-ian) [inline CONTRIBUTING.md:119]:
```suggestion
openJDK which is currently at version 8.0.302. To install that with SDKMAN:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -243,15 +246,15 @@ resources from the resource folders, it won't run any of the scripts that genera
 as compile code to Javascript), in order to do this you can run:
 
 ```bash
-~$ sbt resources
+~$ ./sbt resources
```
Comment (edalex-ian) [inline CONTRIBUTING.md:249]: As mentioned, we want to have consistent style, so remove the shell prompt (here, and also all the others):

```suggestion
./sbt resources
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -335,7 +338,9 @@ EclipseKeys.eclipseOutput := Some("target/scala-2.11/classes/")
 
 #### Code Formatters
 
-All code should be formatted using the following formatting tools:
+All code should be formatted using the following tools.
+Assuming you've followed the above steps to install the pre-commit hooks this will be done automatically for you at commit time.
+However if you don't, then the CI pipelines will fail during checking.
```
Comment (edalex-ian) [inline CONTRIBUTING.md:343]:
```suggestion
However if you don't, then the CI pipelines will fail for your pull requests.
```

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Alright, just a few more tweaks that went astray there in your updates, then tomorrow let's discuss how we can merge the JS part.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -201,31 +267,165 @@ Ensure that your runner settings compiles the whole project before running:
 
 - IntelliJ - `Before Launch -> Build Project`
 
-#### Running the admin tool
+### Running the admin tool
+
+This guide runs sbt in non-interactive mode. You can run in interactive mode to save rebuild time by first running 'sbt', and the another command such as 'compile'.
 
 Ensure you have your server running and know it's
```
Comment (edalex-ian) [inline CONTRIBUTING.md:274]: Hmmm, something has gone astray here. :thinking:

**Review by edalex-ian — CHANGES_REQUESTED** [review]
So so close. Just the 'Running the admin tool' section seems to have gone wacky.

File: CONTRIBUTING.md (hunk 5 above)
Comment (edalex-yinzi) [reply CONTRIBUTING.md:274]: I move this sentence from `build installer` section. I think it just means we can get rid of the init step (those loding info) each time?

Comment (edalex-ian) [reply CONTRIBUTING.md:274]: Okay, how about you change this section to read:

```
### Running the Admin Console

Once you have successfully started the server with the above commands, you can then launch the Admin Console
to manage it. To do so from the command line with SBT, run:
```

Comment (edalex-yinzi) [reply CONTRIBUTING.md:274]: Wired, I got this error when running this command. 

`./sbt compile adminTool/run`

image

Comment (edalex-ian) [reply CONTRIBUTING.md:274]: Oh wow that is odd. That'd be attempting to use javaws perhaps to launch a Java WebStart application - which is what the Admin Console was until only a couple of years ago when we changed it. Strange though, I would've expected that task to run the class...

I reckon leave it in there, but we'll need to raise an issue to look at how that task is written.

Comment (edalex-yinzi) [reply CONTRIBUTING.md]: I asked Penghai to check this command on his laptop, he said it works. So maybe it's my env's issue.

**Review by edalex-ian — APPROVED** [review]
Woohoo! Great to have this doco in better shape. A big leap forward. :+1:

**Pull request merged**